src/visualization/types/SingleStatWithLine/view.tsx

```diff
@@ -161,161 +191,162 @@ const SingleStatWithLine: FC<Props> = ({properties, result, timeRange}) => {
-                style={{backgroundColor}}
-                data-testid="single-stat"
-              >
-                <div className="single-stat--resizer">
-                  <svg
-                    width="100%"
-                    height="100%"
-                    viewBox={`0 0 ${formattedValue.length * 55} 100`}
+    const statPortion = (
+      <LatestValueTransform table={result.table} allowString={true}>
+        {latestValue => {
+          const {
+            bgColor: backgroundColor,
+            textColor,
+          } = generateThresholdsListHexs({
+            colors: properties.colors.filter(c => c.type !== 'scale'),
+            lastValue: latestValue,
+            cellType: 'single-stat',
+          })
+
+          const timeFormatter = getFormatter('time', {
+            timeZone: timeZone === 'Local' ? undefined : timeZone,
+            timeFormat: DEFAULT_TIME_FORMAT,
+          })
+
+          const formattedValue =
+            result.table.getColumnType('_value') === 'time'
+              ? timeFormatter(latestValue)
+              : formatStatValue(latestValue, {
+                  decimalPlaces: properties.decimalPlaces,
+                  prefix: properties.prefix,
+                  suffix: properties.suffix,
+                })
+
+          return (
+            <div
+              className="single-stat"
+              style={{backgroundColor}}
+              data-testid="single-stat"
+            >
+              <div className="single-stat--resizer">
+                <svg
+                  width="100%"
+                  height="100%"
+                  viewBox={`0 0 ${formattedValue.length * 55} 100`}
+                >
+                  <text
+                    className="single-stat--text"
+                    data-testid="single-stat--text"
+                    fontSize="100"
+                    y="59%"
+                    x="50%"
+                    dominantBaseline="middle"
+                    textAnchor="middle"
+                    style={{fill: textColor}}
                   >
-                    <text
-                      className="single-stat--text"
-                      data-testid="single-stat--text"
-                      fontSize="100"
-                      y="59%"
-                      x="50%"
-                      dominantBaseline="middle"
-                      textAnchor="middle"
-                      style={{fill: textColor}}
-                    >
-                      {formattedValue}
-                    </text>
-                  </svg>
-                </div>
+                    {formattedValue}
+                  </text>
+                </svg>
               </div>
-            )
-          }}
-        </LatestValueTransform>
-      </Plot>
+            </div>
+          )
+        }}
+      </LatestValueTransform>
     )
+
+    return <Plot config={config}>{statPortion}</Plot>
+  }
+  // helper functions
+  function makeConfig() {
```

Comment:
I'm not a huge fan of this - creating a method that takes no input and is only called once. I understand the intent, and how it came about (pulling out a big config object used once), but since there's no configuration available via method properties, and since it's only used in one place, all it effectively does is make execute a goto when I'm reading the code that calls `makeConfig`. 

I'd strongly prefer to leave this configuration inline and only pull it out into a method if it's called more than once. I  want to limit premature abstractions as much as possible